### PR TITLE
Add "egl" feature to force use of EGL on OpenGL Linux 

### PR DESCRIPTION
**Pull request by wangcheng22**
1. Add "egl" feature to force use of EGL on OpenGL Linux.  Also rename "android_opengl_context.cpp" and "linux_opengl_context.cpp" to "opengl_egl_context.cpp" and "opengl_glx_context.cpp" as Linux platform can use GLX or EGL.
2. Minor improvement on OpenGL EGL context to query multiple devices for display with EGL_EXT_device_enumeration, EGL_EXT_platform_base.

### CODE REVIEW

File: src/Target.h [hunk 1]
```diff
@@ -82,6 +82,7 @@ struct Target {
         CLHalf = halide_target_feature_cl_half,
         OpenGL = halide_target_feature_opengl,
         OpenGLCompute = halide_target_feature_openglcompute,
+        EGL = halide_target_feature_egl,
```
Comment (steven-johnson) [inline src/Target.h]: This needs to be added to the relevant table in PyEnums.cpp as well

Comment (wangcheng22) [reply src/Target.h]: done

**Comment by steven-johnson** [comment]
LGTM, but want review from @shoaibkamil before landing.

**Comment by steven-johnson** [comment]
Monday morning review ping

**Comment by shoaibkamil** [comment]
Sorry, I'm not at all familiar with EGL, so hopefully @zvookin or @slomp can weigh in.

**Comment by steven-johnson** [comment]
Does anyone else feel qualified to review this? (I don't have enough EGL experience to weigh in)

**Comment by steven-johnson** [comment]
Only failure is unrelated (LLVM 10 issue, see https://github.com/halide/Halide/pull/4021); will land as soon as approved by @zvookin

**Comment by steven-johnson** [comment]
Thanks for the fix!

**Comment by wangcheng22** [comment]
Thanks for all your reviews.

**Pull request merged**